packages/connect-form/src/components/advanced-options-tabs/general/host-input.tsx

```diff
@@ -0,0 +1,148 @@
+import { css } from '@emotion/css';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  Label,
+  Icon,
+  IconButton,
+  TextInput,
+  spacing,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import { ConnectionFormError } from '../../../utils/connect-form-errors';
+import { MARKABLE_FORM_FIELD_NAMES } from '../../../constants/markable-form-fields';
+import DirectConnectionInput from './direct-connection-input';
+import FormFieldContainer from '../../form-field-container';
+
+const hostInputContainerStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  width: '100%',
+  marginBottom: spacing[2],
+});
+
+const hostInputStyles = css({
+  flexGrow: 1,
+});
+
+const hostActionButtonStyles = css({
+  marginLeft: spacing[1],
+  marginTop: spacing[1],
+});
+
+function HostInput({
+  errors,
+  connectionStringUrl,
+  updateConnectionFormField,
+}: {
+  errors: ConnectionFormError[];
+  connectionStringUrl: ConnectionStringUrl;
+  updateConnectionFormField: UpdateConnectionFormField;
+}): React.ReactElement {
+  const [hosts, setHosts] = useState([...connectionStringUrl.hosts]);
+  const { isSRV } = connectionStringUrl;
+
+  const showDirectConnectionInput =
+    connectionStringUrl.searchParams.get('directConnection') === 'true' ||
+    (!connectionStringUrl.isSRV && hosts.length === 1);
+
+  useEffect(() => {
+    // Update the hosts in the state when the underlying connection string hosts
+    // change. This can be when a user changes connections, pastes in a new
+    // connection string, or changes a setting which also updates the hosts.
+    if (connectionStringUrl.hosts !== hosts) {
+      setHosts([...connectionStringUrl.hosts]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [connectionStringUrl]);
```

Comment:
IIUC you disabled the check here because `hosts` is not part of the list, but looking at this code, this check there would always be true, so you might as well just remove it and then you don't need the eslint check disabled. This also matches your comment and this effect a bit better than with a check present I think:

```suggestion
  useEffect(() => {
    // Update the hosts in the state when the underlying connection string hosts
    // change. This can be when a user changes connections, pastes in a new
    // connection string, or changes a setting which also updates the hosts.
    setHosts([...connectionStringUrl.hosts]);
  }, [connectionStringUrl]);
```

Reply:
Ah ty yup this check was definitely not needed. removed